### PR TITLE
LCS-654 incompleted courses fetch by days

### PR DESCRIPTION
**Pull request by mmackiewicz1**
Catalogue service now returns required courses according to due dates.

### CODE REVIEW

File: src/main/java/uk/gov/cslearning/catalogue/api/CourseController.java [hunk 1]
```diff
@@ -28,12 +35,27 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
```
Comment (willh) [inline src/main/java/uk/gov/cslearning/catalogue/api/CourseController.java:58]: Can we collapse some of these using statements with wildcard imports without conflict? e.g. `import org.springframework.web.bind.annotation.*`

Comment (mmackiewicz1) [reply src/main/java/uk/gov/cslearning/catalogue/api/CourseController.java:58]: I would not recommend this. Importing with "*" causes all libraries to be imported and compiled. It looks neat in code but is actually not memory efficient.

Comment (willh) [reply src/main/java/uk/gov/cslearning/catalogue/api/CourseController.java:65]: Yes, at compile time, but once the linker runs and determines there's no actual reference to the other imported classes they won't be included in the generated bytecode, surely? (This conversation is a bit academic so to be clear I'm not interested in blocking merge over it)

**Review by mattdoherty-kainos — APPROVED** [review]
lgtm

**Pull request merged**